### PR TITLE
ARROW-17927: [C++] Changed SleepABitAsync to use a thread pool to reduce the # of running threads

### PR DESCRIPTION
**Pull request by westonpace**

### CODE REVIEW

**Comment by westonpace** [comment]
There are already two alternatives at #14332 and #14334 but I feel this might be simpler while still achieving the same goals.

**Comment by westonpace** [comment]
@github-actions crossbow submit -g cpp

File: cpp/src/arrow/testing/gtest_util.cc [hunk 1]
```diff
@@ -767,22 +769,20 @@ void BusyWait(double seconds, std::function<bool()> predicate) {
...
-    SleepFor(seconds);
-    out.MarkFinished();
-  }).detach();
-  return out;
+// These threads will spend most of their time sleeping so there
+// is no need to base this on the # of cores.  Instead it should be
+// high enough to ensure good concurrency when there is concurrent hardware
+static constexpr int kNumSleepThreads = 32;
+
+std::shared_ptr<ThreadPool> CreateSleepThreadPool() {
```
Comment (pitrou) [inline cpp/src/arrow/testing/gtest_util.cc:777]: Nit: `static`?

**Review by pitrou — APPROVED** [review]
LGTM. Nice that we found a simple solution after all :-)

**Pull request merged**